esphome/components/uart/uart_component_rp2040.cpp

```diff
@@ -52,11 +52,19 @@ uint16_t RP2040UartComponent::get_config() {
 }
 
 void RP2040UartComponent::setup() {
-  if (this->rx_pin_) {
-    this->rx_pin_->setup();
-  }
-  if (this->tx_pin_ && this->rx_pin_ != this->tx_pin_) {
-    this->tx_pin_->setup();
+  auto setup_component_if_needed = [](InternalGPIOPin *pin) {
+    if (!pin) {
+      return;
+    }
+    const auto mask = gpio::Flags::FLAG_OPEN_DRAIN | gpio::Flags::FLAG_PULLUP | gpio::Flags::FLAG_PULLDOWN;
+    if (pin->get_flags() & mask) {
```

Comment:
```suggestion
    if ((pin->get_flags() & mask) != gpio::Flags::FLAG_NONE) {
```

libretiny version is a bit more explict and clearer about the intent